File: Design_Pattern_and_Principles/ObserverPatternExample/StockMarket.java
import java.util.ArrayList;

public class StockMarket implements Stock {
    private ArrayList<Observer> observers;
    private String stockSymbol;
    private double price;

    public StockMarket() {
        observers = new ArrayList<>();
    }

    public void registerObserver(Observer o) {
        observers.add(o);
    }

    public void removeObserver(Observer o) {
        observers.remove(o);
    }

    public void updatePrice(String stockSymbol, double price) {
        this.stockSymbol = stockSymbol;
        this.price = price;
        notifyObservers();
    }

    public void notifyObservers() {
        for (Observer o : observers) {
            o.update(stockSymbol, price);
        }
    }
}

File: Design_Pattern_and_Principles/ObserverPatternExample/StockMarket.cs
using System.Collections.Generic;

public class StockMarket : IStock {
    private List<IObserver> observers;
    private string stockSymbol;
    private double price;

    public StockMarket() {
        observers = new List<IObserver>();
    }

    public void RegisterObserver(IObserver o) {
        observers.Add(o);
    }

    public void RemoveObserver(IObserver o) {
        observers.Remove(o);
    }

    public void UpdatePrice(string stockSymbol, double price) {
        this.stockSymbol = stockSymbol;
        this.price = price;
        NotifyObservers();
    }

    public void NotifyObservers() {
        foreach (IObserver o in observers) {
            o.Update(stockSymbol, price);
        }
    }
}